src/main/java/de/medizininformatikinitiative/flare/model/fhir/Reference.java

```diff
@@ -2,15 +2,9 @@
 
 import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
 
-import java.util.Objects;
-
 @JsonIgnoreProperties(ignoreUnknown = true)
 public record Reference(String reference) {
 
-    public Reference {
-        Objects.requireNonNull(reference);
-    }
-
     public String id() {
         return reference.substring(reference.indexOf('/') + 1);
```

Comment:
Because the reference can be null now, We have to return an `Optional<String>` from `id()`.